broker/lib/controllers/FabrikBaseController.js

```diff
@@ -30,6 +33,71 @@ class FabrikBaseController extends BaseController {
     return _.get(config, 'broker_name', 'service-fabrik-broker');
   }
 
+  handleByLockingResource(func, operationType, lastOperationCall) {
+    return (req, res, next) => {
+      return Promise.try(() => {
+          if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR && config.enable_service_fabrik_v2) {
+            // Acquire lock for this instance
+            return lockManager.lock(req.params.instance_id, {
+                lockedResourceDetails: {
+                  resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+                  resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+                  resourceId: req.params.instance_id,
+                  operation: operationType ? operationType : req.query.operation.type // This is for the last operation call
+                }
+              })
+              .catch(err => {
+                if (lastOperationCall && err instanceof ResourceAlreadyLocked) {
+                  logger.info(`Proceeding as lock is already acquired for the resource: ${req.params.instance_id}`);
+                } else {
+                  return next(err);
+                }
+              });
+          }
+        })
+        .then(() => this._handleWithUnlock(func, operationType, lastOperationCall, req, res, next));
+    };
+  }
+
+  _handleWithUnlock(func, operationType, lastOperationCall, req, res, next) {
```

Comment:
Minor comment: Can this function be renamed to something more understandable?

Reply:
Changed to `_handleWithResourceUnlock`